### DISCUSSION TITLE
CCSGOUserCmdPB *Crash Fix*

### DISCUSSION THREAD

**Discussion by siongadev** [maecry/asphyxia-cs2#12] [General]
If you were crashing due to this
![image](https://github.com/maecry/asphyxia-cs2/assets/121215919/c1905cf9-3d29-44c3-9cec-e275ec795fc8)

please replace with 

class CCSGOUserCmdPB
{
public:
    int32_t nTickCount; // 0x0
    MEM_PAD(0x4); // 0x4
    void* pInputHistory; // 0x8

    CCSGOInputHistoryEntryPB* GetInputHistoryEntry(std::int32_t nTick)
    {
        if (nTick < this->nTickCount && pInputHistory)
        {
            // Ensure that pInputHistory is a valid pointer before using it.
            // Assuming arrTickList is an array of pointers.
            CCSGOInputHistoryEntryPB** arrTickList = reinterpret_cast<CCSGOInputHistoryEntryPB**>(pInputHistory);

            if (arrTickList)
            {
                // Verify that nTick is within bounds.
                if (nTick >= 0 && nTick < this->nTickCount)
                {
                    return arrTickList[nTick];
                }
            }
        }

        return nullptr;
    }
};

^ This new version should work if not let me know and I'll fix

**Comment by siongadev** [comment]
Edit: if the copy+paste is weird use this link

https://pastebin.com/raw/C8f4sS3b

**Comment by maecry** [comment]
`if (nTick >= 0 && nTick < this->nTickCount)`
This check is unnecessary as we did above lol, will push a fix soon
